### PR TITLE
Set MAKE env var

### LINKED ISSUE #590: built make tool does not get put into $PATH for configure_make() projects
Symptom:

On a system without a preinstalled `make`:

```
% cd examples/third_party
% bazel build @bison//:bison
..
config.status: executing depfiles commands
config.status: error: in `/root/.cache/bazel/_bazel_root/c319f04a519246939a722b6a9cd3d55c/sandbox/processwrapper-sandbox/6/execroot/rules_foreign_cc_examples_third_party/bazel-out/k8-fastbuild/bin/external/bison/bison.build_tmpdir':
config.status: error: Something went wrong bootstrapping makefile fragments
    for automatic dependency tracking.  Try re-running configure with the
    '--disable-dependency-tracking' option to at least be able to build
    the package (albeit without support for automatic dependency tracking).
See `config.log' for more details
_____ END BUILD LOGS _____
rules_foreign_cc: Build wrapper script location: bazel-out/k8-fastbuild/bin/external/bison/bison_foreign_cc/wrapper_build_script.sh
rules_foreign_cc: Build script location: bazel-out/k8-fastbuild/bin/external/bison/bison_foreign_cc/build_script.sh
rules_foreign_cc: Build log location: bazel-out/k8-fastbuild/bin/external/bison/bison_foreign_cc/Configure.log

Target @bison//:bison failed to build
Use --verbose_failures to see the command lines of failed build steps.
INFO: Elapsed time: 27.618s, Critical Path: 25.23s
INFO: 2 processes: 2 internal.
FAILED: Build did NOT complete successfully

% grep -C 8 'line 2792' /root/.cache/bazel/_bazel_root/c319f04a519246939a722b6a9cd3d55c/sandbox/processwrapper-sandbox/6/execroot/rules_foreign_cc_examples_third_party/bazel-out/k8-fastbuild/bin/external/bison/bison.build_tmpdir/config.log
config.status:2462: creating tests/bison
config.status:2462: creating Makefile
config.status:2462: creating po/Makefile.in
config.status:2462: creating doc/yacc.1
config.status:2462: creating lib/config.h
config.status:2693: linking /root/.cache/bazel/_bazel_root/c319f04a519246939a722b6a9cd3d55c/sandbox/processwrapper-sandbox/6/execroot/rules_foreign_cc_examples_third_party/external/bison/GNUmakefile to GNUmakefile
config.status:2712: executing depfiles commands
config.status:2789: cd .       && sed -e '/# am--include-marker/d' Makefile         | make -f - am--depfiles
./config.status: line 2792: make: command not found
config.status:2794: $? = 127
config.status:2798: error: in `/root/.cache/bazel/_bazel_root/c319f04a519246939a722b6a9cd3d55c/sandbox/processwrapper-sandbox/6/execroot/rules_foreign_cc_examples_third_party/bazel-out/k8-fastbuild/bin/external/bison/bison.build_tmpdir':
config.status:2800: error: Something went wrong bootstrapping makefile fragments
    for automatic dependency tracking.  Try re-running configure with the
    '--disable-dependency-tracking' option to at least be able to build
    the package (albeit without support for automatic dependency tracking).
See `config.log' for more details


```

The local `make` is built correctly.  But the `children_to_path` shell function does not add `$EXT_BUILD_DEPS/bin/make/bin` to the `PATH`, so `configure` cannot find the freshly built `make` (non-absolute path).

```
bash-4.2# ls -l /root/.cache/bazel/_bazel_root/c319f04a519246939a722b6a9cd3d55c/sandbox/processwrapper-sandbox/6/execroot/rules_foreign_cc_examples_third_par
ty/bazel-out/k8-fastbuild/bin/external/bison/bison.ext_build_deps/bin/
total 4
drwxr-xr-x 5 root root 4096 Mar 29 16:33 make
bash-4.2# ls -l /root/.cache/bazel/_bazel_root/c319f04a519246939a722b6a9cd3d55c/sandbox/processwrapper-sandbox/6/execroot/rules_foreign_cc_examples_third_party/bazel-out/k8-fastbuild/bin/external/bison/bison.ext_build_deps/bin/make/
total 12
drwxr-xr-x 2 root root 4096 Mar 29 16:33 bin
drwxr-xr-x 2 root root 4096 Mar 29 16:33 include
drwxr-xr-x 5 root root 4096 Mar 29 16:33 share
bash-4.2# ls -l /root/.cache/bazel/_bazel_root/c319f04a519246939a722b6a9cd3d55c/sandbox/processwrapper-sandbox/6/execroot/rules_foreign_cc_examples_third_party/bazel-out/k8-fastbuild/bin/external/bison/bison.ext_build_deps/bin/make/bin/
total 4
lrwxrwxrwx 1 root root 213 Mar 29 16:33 make -> /root/.cache/bazel/_bazel_root/c319f04a519246939a722b6a9cd3d55c/sandbox/processwrapper-sandbox/6/execroot/rules_foreign_cc_examples_third_party/bazel-out/host/bin/external/rules_foreign_cc/toolchains/make/bin/make
bash-4.2#
```

### PR DESCRIPTION
**Pull request by jsharpe**
Rework of patch in #549. Fixes #590.

### CODE REVIEW

File: foreign_cc/private/configure_script.bzl [hunk 1]
```diff
@@ -41,6 +42,8 @@ def create_configure_script(
         root_path = "$$BUILD_TMPDIR$$"
         configure_path = "{}/{}".format(root_path, configure_command)
 
+    script.append("export MAKE={}".format(make_path))
```
Comment (UebelAndre) [inline foreign_cc/private/configure_script.bzl:45]:
```suggestion
    script.append("##export_var## MAKE {}".format(make_path))
```

**Review by UebelAndre — CHANGES_REQUESTED** [review]
I think this is a fine change but I wonder if we could solve for the same thing by fixing the way we inject things into `PATH`. Would that solve for the same thing?

**Comment by jsharpe** [comment]
> I think this is a fine change but I wonder if we could solve for the same thing by fixing the way we inject things into `PATH`. Would that solve for the same thing?

No, because that would rely on the tool in the make toolchain being called "make" whereas setting the `MAKE` environment variable allows tools to be named differently e.g. `gmake` is a common name for installing GNUMake on systems that have BSD make under the name `make`

**Comment by UebelAndre** [comment]
@googlebot I fixed it.

**Comment by UebelAndre** [comment]
@googlebot I fixed it.

**Comment by UebelAndre** [comment]
I'm not sure what's up with CLA but @jsharpe and I have definitely signed it.

**Pull request merged**